src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,118 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    public static final int DEFAULT_CAPACITY = 10;
+    public static final double ARRAY_GROWTH_VALUE = 1.5;
+    private Object[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        growIfArrayIsFull();
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexForAdd(index);
+        growIfArrayIsFull();
+        System.arraycopy(elementData, index, elementData, index + 1,
+                size - index);
+        elementData[index] = value;
+        size++;
+    }
+
+    private void checkIndexForAdd(int index) {
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Can't add value to index " + index
+                    + ". Index is invalid.");
+        }
+    }
 
+    private void growIfArrayIsFull() {
+        if (size == elementData.length) {
+            Object[] grownElementData = new Object[(int) (elementData.length
+                                + (elementData.length * 1.5))];
+            System.arraycopy(elementData, 0, grownElementData, 0, size);
+            elementData = grownElementData;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndexValidation(index);
+        return (T) elementData[index];
+    }
+
+    private void checkIndexValidation(int index) {
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("Index is invalid.");
```

Comment:
The `ArrayListIndexOutOfBoundsException` class is missing. You need to define this custom exception to handle invalid indices as per the task requirements.